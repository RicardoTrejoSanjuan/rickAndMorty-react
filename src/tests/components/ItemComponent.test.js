import React from 'react';
const { shallow } = require("enzyme");
const { default: ItemComponent } = require("../../components/ItemComponent");

describe('Test en <ItemComponent />', () => {

    const character={
        id: 1,
        name: "Rick Sanchez",
        status: "Alive",
        species: "Human",
        gender: "Male",
        image: "https://rickandmortyapi.com/api/character/avatar/1.jpeg",
    };
    
    test('Dede mostrar el componente', () => {
        const wrapper = shallow(<ItemComponent character={character} />);

        expect(wrapper).toMatchSnapshot();
    });

    test('Debe contener un <div className="card">', () => {
        const wrapper = shallow(<ItemComponent character={character} />);

        expect(wrapper.find('.card').exists()).toBe(true);
    });

});