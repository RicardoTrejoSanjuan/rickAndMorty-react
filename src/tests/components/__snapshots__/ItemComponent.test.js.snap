// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test en <ItemComponent /> Dede mostrar el componente 1`] = `
<div
  className="card-box animate__animated animate__bounce animate__jello"
>
  <div
    className="card"
  >
    <div
      className="cara"
    >
      <img
        alt="Rick Sanchez"
        src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
      />
    </div>
    <div
      className="cara info"
    >
      <div
        className="containerImg"
      >
        <div
          className="titleCard"
        >
          <h5>
            <b>
              Rick Sanchez
            </b>
          </h5>
        </div>
        <p>
          <b>
            Species:
          </b>
           
          Human
        </p>
        <p>
          <b>
            Status:
          </b>
           
          Alive
        </p>
        <p>
          <b>
            Gender:
          </b>
           
          Male
        </p>
      </div>
    </div>
  </div>
</div>
`;
